File: app/src/main/java/com/boardgamegeek/service/CollectionHasPartsUploadTask.java
package com.boardgamegeek.service;

import android.support.annotation.NonNull;

import com.boardgamegeek.provider.BggContract.Collection;
import com.boardgamegeek.service.model.CollectionItem;

import okhttp3.OkHttpClient;

public class CollectionHasPartsUploadTask extends CollectionTextUploadTask {
	public CollectionHasPartsUploadTask(OkHttpClient client) {
		super(client);
	}

	@Override
	@NonNull
	protected String getTextColumn() {
		return Collection.HASPARTS_LIST;
	}

	@Override
	public String getTimestampColumn() {
		return Collection.HAS_PARTS_DIRTY_TIMESTAMP;
	}

	@Override
	@NonNull
	protected String getFieldName() {
		return "haspartslist";
	}

	@Override
	protected String getValue(CollectionItem collectionItem) {
		return collectionItem.getHasParts();
	}

	@Override
	public boolean isDirty() {
		return collectionItem.getHasPartsDirtyTimestamp() > 0;
	}
}

File: app/src/main/java/com/boardgamegeek/service/CollectionHasPartsUploadTask.kt
package com.boardgamegeek.service

import com.boardgamegeek.provider.BggContract.Collection
import okhttp3.OkHttpClient

class CollectionHasPartsUploadTask(client: OkHttpClient) : CollectionTextUploadTask(client) {
    override val timestampColumn = Collection.Columns.HAS_PARTS_DIRTY_TIMESTAMP

    @Suppress("SpellCheckingInspection")
    override val fieldName = "haspartslist"

    override fun getValue() = collectionItem.hasParts.orEmpty()

    override val isDirty: Boolean
        get() = collectionItem.hasPartsDirtyTimestamp > 0
}
